### PR TITLE
ExtensionInfo loader replacement

### PR DESCRIPTION
**Pull request by MaowImpl**
The foundation for the ExtensionInfo loader class, currently broken.

The purpose of replacing ExtensionInfo's loader is to allow for custom executors so that people can add extension-specific configurations, e.g. `<NoPortHack/>` (just an example). The code is a little messy and could be improved, but I think the higher priority is getting it to work.

### CODE REVIEW

**Comment by MaowImpl** [comment]
Fixed and modified so that each nested tag in `HacknetExtension` would have its own executor. Less performant, but it's now more extensible.

**Comment by Fayti1703** [comment]
Is there a reason you're removing `BaseGameFixes/NoToLowerOnTheme.cs` in this PR?

**Comment by MaowImpl** [comment]
`BaseGameFixes/NoToLowerOnTheme.cs` only applies to the method I'm replacing.
I don't call a `.ToLower()` on the theme path, so there's no point in keeping that patch.

File: PathfinderAPI/Replacements/ExtensionInfoLoader.cs [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+                    throw new FormatException($"Invalid language in '{filepath}': {language}");
+                extInfo.Language = language;
+            });
+            exe.Reg("AllowSaves", (exec, info) => extInfo.AllowSave = info.ContentAsBoolean(filepath));
+            exe.Reg("StartingVisibleNodes", (exec, info) =>
+                extInfo.StartingVisibleNodes = info.Content
+                    .Replace(@"\s+", "")
+                    .Split(','));
+            exe.Reg("StartingMission", (exec, info) => extInfo.StartingMissionPath = info.Content);
+            exe.Reg("StartingActions", (exec, info) => extInfo.StartingActionsPath = info.Content);
```
Comment (Windows10CE) [inline PathfinderAPI/Replacements/ExtensionInfoLoader.cs:129]: Both StartingMission and StartingActions should evaluate as null when `NONE` is passed in, see the related base game code:
```cs
				if (xmlReader.Name == "StartingMission")
				{
					xmlReader.MoveToContent();
					extensionInfo.StartingMissionPath = xmlReader.ReadElementContentAsString();
					if (extensionInfo.StartingMissionPath == "NONE")
					{
						extensionInfo.StartingMissionPath = null;
					}
				}
				if (xmlReader.Name == "StartingActions")
				{
					xmlReader.MoveToContent();
					extensionInfo.StartingActionsPath = xmlReader.ReadElementContentAsString();
					if (extensionInfo.StartingActionsPath == "NONE")
					{
						extensionInfo.StartingActionsPath = null;
					}
				}
```

File: PathfinderAPI/Replacements/ExtensionInfoLoader.cs [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+            {
+                var language = info.Content;
+                if (!ValidLanguages.Contains(language))
+                    throw new FormatException($"Invalid language in '{filepath}': {language}");
+                extInfo.Language = language;
+            });
+            exe.Reg("AllowSaves", (exec, info) => extInfo.AllowSave = info.ContentAsBoolean(filepath));
+            exe.Reg("StartingVisibleNodes", (exec, info) =>
+                extInfo.StartingVisibleNodes = info.Content
+                    .Replace(@"\s+", "")
```
Comment (Windows10CE) [inline PathfinderAPI/Replacements/ExtensionInfoLoader.cs:126]: `string.Replace(string, string)` does not parse regex, this line would only remove occurrences of the literal string `\s+`.

**Review by Windows10CE — CHANGES_REQUESTED** [review]

**Review by Windows10CE — APPROVED** [review]

**Pull request merged**